### PR TITLE
RSocket Proxy for Prometheus monitoring

### LINKED ISSUE #122: Showcase SCSt 3.0 + Micrometer + RSocket Proxy for Prometheus monitoring
As we moved away from the older service-discovery to the new RSocket proxy architecture, it'd be useful to highlight the specific dependencies with the help of custom SCSt (3.0) applications.

**Acceptance:**
- Rename `spring-cloud-dataflow-samples/tree/master/micrometer` to `spring-cloud-dataflow-samples/tree/master/micrometer-samples`
- Rename `spring-cloud-dataflow-samples/tree/master/micrometer/tasks` to `spring-cloud-dataflow-samples/tree/master/micrometer-samples/task-apps`
- Describe or cross-link the description of the new architecture in the root README
- Highlight the _required_ Micrometer Commons (from app starters BOM) + RSocket Proxy dependencies
- Add the apps under `spring-cloud-dataflow-samples/tree/master/micrometer/stream-apps`
- Update ref. docs, samples, and the microsite references to the old links

### PR DESCRIPTION
**Pull request by tzolov**
- Rename spring-cloud-dataflow-samples/tree/master/micrometer to spring-cloud-dataflow-samples/tree/master/micrometer-samples
 - Rename spring-cloud-dataflow-samples/tree/master/micrometer/tasks to spring-cloud-dataflow-samples/tree/master/micrometer-samples/task-apps
 - Add the apps under spring-cloud-dataflow-samples/tree/master/micrometer/stream-apps
 - Describe or cross-link the description of the new architecture in the root README

Resolves #122

### CODE REVIEW

File: monitoring-samples/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Spring Cloud Data Flow Monitoring - Prometheus RSocket Proxy Architecture
```
Comment (sabbyanandan) [inline monitoring-samples/README.md]: > Monitoring Architecture: Spring Cloud Data Flow + Prometheus RSocket Proxy

File: monitoring-samples/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+The [stream-apps](./stream-apps/) samples shows how to enable monitoring for custom built `source`, `processor` and `sink` apps.  
+
+The [task-apps](./task-apps/) sample shows how to enable monitoring for custom built `task` apps.
+
+The following image shows the general architecture of how the Stream applications are monitored:
+
+![Stream App Monitoring](./images/SCDF-monitoring-prometheus-streams.png)
+
+and the same architecture is used for monitoring tasks as wel.
```
Comment (sabbyanandan) [inline monitoring-samples/README.md]: > and the same architecture is used for monitoring the tasks as well.

File: monitoring-samples/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+The prebuilt `stream` and `task` applications support the Prometheus Proxy RSocket monitoring architecture. 
+
+The [stream-apps](./stream-apps/) samples shows how to enable monitoring for custom built `source`, `processor` and `sink` apps.  
+
+The [task-apps](./task-apps/) sample shows how to enable monitoring for custom built `task` apps.
+
+The following image shows the general architecture of how the Stream applications are monitored:
+
+![Stream App Monitoring](./images/SCDF-monitoring-prometheus-streams.png)
```
Comment (sabbyanandan) [inline monitoring-samples/README.md]: "lifecycle management" reference in the diagram is confusing. We should clarify it as "facilitate lifecycle management for stream apps" or something like that.

File: monitoring-samples/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+and the same architecture is used for monitoring tasks as wel. 
+ 
+![Stream App Monitoring](./images/SCDF-monitoring-prometheus-tasks.png)
```
Comment (sabbyanandan) [inline monitoring-samples/README.md]: "Grafana links shortcuts" <- not sure I understand that .. just describe it as a clickable button instead

File: monitoring-samples/stream-apps/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+		<dependency>
+			<groupId>org.springframework.cloud</groupId>
+			<artifactId>spring-cloud-starter-stream-rabbit</artifactId>
+		</dependency>
+
+
+		<dependency>
+			<groupId>org.springframework.cloud.stream.app</groupId>
+			<artifactId>app-starters-micrometer-common</artifactId>
```
Comment (sabbyanandan) [inline monitoring-samples/stream-apps/README.md]: Should we describe what they dependency provides somewhere?

File: monitoring-samples/README.md (hunk 4 above)
Comment (tzolov) [reply monitoring-samples/README.md]: "clickable button" is bad as well.   What about "UI interface to access the SCDF dashboards" or just to leave it empty?

Comment (sabbyanandan) [reply monitoring-samples/README.md]: Yeah, empty would be better.

File: monitoring-samples/stream-apps/README.md [hunk 6]
```diff
@@ -1,37 +1,41 @@
...
-			<groupId>io.micrometer.prometheus</groupId>
-			<artifactId>prometheus-rsocket-spring</artifactId>
-			<version>0.9.0</version>
-		</dependency>
-		...
-	</dependencies>
+```xml
+    <dependency>
+        <groupId>org.springframework.cloud.stream.app</groupId>
+        <artifactId>app-starters-micrometer-common</artifactId>
```
Comment (sabbyanandan) [inline monitoring-samples/stream-apps/README.md]: The `app-starters-micrometer-common` purpose is still not described. Perhaps a one-liner about what this adds to the applications would help. Esp, this is a required dependency if users would have to use the monitoring support in SCDfF.

Comment (sabbyanandan) [reply monitoring-samples/stream-apps/README.md]: Oh, nevermind! You have it described below.

File: monitoring-samples/stream-apps/README.md [hunk 7]
```diff
@@ -1,37 +1,41 @@
...
+    <dependency>
+        <groupId>io.micrometer.prometheus</groupId>
+        <artifactId>prometheus-rsocket-spring</artifactId>
+        <version>0.9.0</version>
+    </dependency>
 ```
 
-For Kafka binder replace `spring-cloud-starter-stream-rabbit` by `spring-cloud-starter-stream-kafka` instead
+The `micrometer-registry-prometheus` amd `prometheus-rsocket-spring` activate the Micrometer Prometheus support for RSocket Proxy meter registry.
+The `app-starters-micrometer-common` would inject some common metrics Tags such as `stream.name`, `stream.app.label`, `application.type` and `application.guid`.
```
Comment (sabbyanandan) [inline monitoring-samples/stream-apps/README.md]: Maybe we could describe it in the context of SCDF. Or even in the context of the OOTB Grafana Dashboard that we ship, and why these tags are important for the PromQL.

Comment (tzolov) [reply monitoring-samples/stream-apps/README.md]: the tags set by the app-starters-micrometer-common a generic for all ~15 TSDBs supported by micrometer  including Prometheus and Influx DB.  We use them for the Influx Dashboards as well. They can be used to aggregate metrics by stream, apps and app instance. For example we used them to implement the Rate-delta alert in the scaling demo. 
Tags are briefly explained in the microsite [here](https://dataflow.spring.io/docs/2.3.0.SNAPSHOT/feature-guides/streams/monitoring/) (btw i need to update the diagrams there as well)
Also imo, the focus of this README is on how to make it work even if i don't understand it. E.g. the goal is to plug a custom Stream App in the existing SCDF monitoring architecture. The detailed explanation of the architecture should be in the microsite?

**Comment by tzolov** [comment]
Rebased squashed and merged at f8078ed08f3c0e22fab14877a812f39536773600

**Pull request closed without merging**